wandb/board/ui/src/pages/Run.js

```diff
@@ -25,14 +25,36 @@ class Run extends React.Component {
 
   componentDidUpdate() {
     window.Prism.highlightAll();
+    if (!this.props.loading) {
+      this.props.refetch({detailed: true});
+    }
   }
 
   componentWillReceiveProps(nextProps) {
+    if (nextProps.model && !_.isEqual(nextProps.model, this.props.model)) {
+      this.setState({model: nextProps.model, bucket: nextProps.bucket});
+    } else if (!nextProps.model) {
+      try {
+        let query = nextProps.client.readQuery({
+          query: RUNS_QUERY,
+          variables: {
+            entityName: nextProps.match.params.entity,
+            name: nextProps.match.params.model,
+            order: 'timeline',
+          },
+        });
+        let model = _.find(query.model.buckets.edges, function(item) {
+          return item.node.name === nextProps.match.params.run;
+        });
+        this.setState({model: query.model, bucket: model.node});
+      } catch (e) {}
```

Comment:
What are we catching here?

Reply:
Also, why are we doing a `RUNS_QUERY` on the run page?  I don't like this `setState` approach.  We should be getting model and bucket from the `MODEL_QUERY`, and `detailed: true` in DidUpdate should load further details.  I don't understand what the state approach gives us.

Reply:
Perhaps we just have a state attribute like `"detailLoaded"` which can be set after the refetch finishes.

Reply:
But it seems like checking for `bucket.history` is sufficient.

Reply:
As `readQuery` doesn't have error handler, try ... catch is needed for script not to break.

Idea with `RUNS_QUERY` is to load cached basic data for Run, from Runs page. In that case, no overlay loader will be displayed at all, just segment loaders for chart and logs. And state would be single source of data for model and bucket.
Other idea was to use `readFragment` for `BasicRunFragment`, but fragment id is not available at this point on Run page.

But, without reading cached data, page will still load much faster.

Checking `bucket.history` is here to distinct if Views is used on Run page, or Runs list. If on individual Run page, `history` parameter will be used.

Reply:
Understood, let's refactor this to use readFragment.  I just pushed a change to master that lets you easily generate the id.  A bucket id looks like: `"BucketType:Entity:wandb:Model:test:Bucket:abc123"` and is then base 64 encoded.  So you can create a helper to generate one from params.  Something like:

```js
function gid(params) {
   return btoa(["BucketType", "Entity", params.entity, "Model", params.model, "Bucket", params.run_id].join(":"))
}
```

Then the fragment approach should work, let me know if it doesn't!

Reply:
Also, the global Apollo error middleware should catch the error and handle it I believe...